File: C#(VisualStudio)/WorkSpace_ASP.NETCoreMVC/書籍検索アプリ_移植元Java/Models/DatabaseConnect.cs
using MySql.Data.MySqlClient;

namespace 書籍検索アプリ_移植元Java.Models
{
    public class DatabaseConnect
    {
        //DB接続用定数
        private const string SERVER_NAME = "localhost";
	    private const string DATABASE_NAME = "booksearchsystem";
	    private const string PROPATIES = "Charset=utf8;SslMode=none;AllowPublicKeyRetrieval=true;";
        //DB接続用・ユーザ定数
	    private const string USER = "root";
	    private const string PASS = "12345";

        // "Server=localhost;Database=booksearchsystem;User Id=aaaa;Password=aaaa;Charset=utf8;SslMode=none;AllowPublicKeyRetrieval=true;";
        private const string connectionString = $"Server={SERVER_NAME};Database={DATABASE_NAME};" +
            $"User Id={USER};Password={PASS};{PROPATIES}";

        //----------
	    //コンストラクタ
	    public DatabaseConnect() {
	    }

        public MySqlCommand GetConnection(string sql){
		
            using var conn = new MySqlConnection(connectionString);
            conn.Open();

            using var cmd = new MySqlCommand(sql, conn);

            return cmd;
        }
    }
}

File: 3_Java(eclipse)/1_workspace_ServletAndJSP/1_書籍検索アプリ/src/main/java/todo/model/DatabaseConnect.java
package todo.model;

import java.sql.Connection;
import java.sql.DriverManager;
import java.sql.SQLException;

public class DatabaseConnect {

	//DB接続用定数
	private static final String DATABASE_NAME = "booksearchsystem";
	private static final String PROPATIES = "?characterEncoding=UTF-8&useSSL=false&allowPublicKeyRetrieval=true";
	private static final String URL = "jdbc:mysql://localhost/" + DATABASE_NAME + PROPATIES;
	//DB接続用・ユーザ定数
	private static final String USER = "root";
	private static final String PASS = "12345";
	
	//----------
	//コンストラクタ
	public DatabaseConnect() {
	}
	
	//古い環境でも動かせるようにするおまじない
	static {
		
	    try {
	    	
	        Class.forName("com.mysql.cj.jdbc.Driver");
	    } catch (ClassNotFoundException e) {
	    	
	        throw new RuntimeException("MySQL JDBC Driver not found", e);
	    }
	}
	
	public Connection getConnection() throws SQLException {
		
        return DriverManager.getConnection(URL, USER, PASS);
    }
}